src/components/SearchBar/SearchCourse/index.js

```diff
@@ -0,0 +1,220 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { useLocation } from "react-router-dom";
+import { useHistory } from "react-router-dom";
+import { actions as courseActions } from "../../Course/redux/action";
+import { actions as pathwayActions } from "../../PathwayCourse/redux/action";
+import SearchBar from "..";
+import { breakpoints } from "../../../theme/constant";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import { Link } from "react-router-dom";
+import { PATHS, interpolatePath } from "../../../constant";
+import {
+  Box,
+  TextField,
+  Container,
+  Typography,
+  Grid,
+  Card,
+} from "@mui/material";
+import useStyles from "../styles";
+import Tooltip from "@mui/material/Tooltip";
+
+function SearchCourse(props) {
+  //   console.log("props", props);
+  const { data } = useSelector(({ Course }) => Course);
+  const pathway = useSelector((state) => state.Pathways);
+  const dispatch = useDispatch();
+  const query = new URLSearchParams(useLocation().search).get("search");
+  const [search, setSearch] = useState(query ? query : "");
+  const history = useHistory();
+  const classes = useStyles();
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+
+  useEffect(() => {
+    dispatch(courseActions.getCourses());
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(pathwayActions.getPathways());
+  }, [dispatch]);
+
+  const handleSearchChange = (e) => {
+    history.push(`?search=${e.target.value}`);
```

Comment:
If a person types in p, y, t, h, o, then n, then they will have ?search=p, ?search=py, ?search=pyt, ?search=pyth, ?search=pytho, ?search=python in their history.  Is this desirable?  Also, if they hit the back button, the matches won't update.  So for example, if the user goes back to the page with the search for p, they'll still see the results for python instead of any course with a p.